### ISSUE TITLE
ML guidance documentation

### ISSUE THREAD

**Issue by louisfh** [kitzeslab/opensoundscape#483]
We should add a guidance on ML model interpretation and practical considerations to the documentation.

More ideas for general machine learning concepts to add to docs:

* Sam's glossary
* Sam's OPSO/TF basics notebook
* Sam's links:
  * https://jalammar.github.io/visual-interactive-guide-basics-neural-networks/
  * https://jalammar.github.io/feedforward-neural-networks-visual-interactive/
  * https://victorzhou.com/blog/intro-to-neural-networks/
  * https://victorzhou.com/blog/intro-to-cnns-part-1/
* What are the conceptual differences between training and validation?
* Why do augmentation on the validation data?
* Validation set might give an inflated estimate of generalizability/performance if training and validation data came from same dataset (e.g. same field site) and especially from same source files (e.g. a training sample from 0,5s and a validation sample from 5,10s of the same file). Could avoid this, e.g. using temporal partitioning to avoid including validation files from the same training data
* Why batch size matters, especially with batch-normalizing algorithms
* Metrics for memory/processor usage and why they matter (e.g. batch size)
* Tips for selecting lengths of audio clips and dimensions of image
* Always look at the inputs to CNNs (e.g. by viewing samples from a preprocessor) to see if they look reasonable.

_Originally posted by @rhine3 in https://github.com/kitzeslab/opensoundscape/issues/335#issuecomment-888547898_

**Comment by louisfh** [comment]
Other points taken from the original issue:
- Training model –  show how to plot validation loss alongside training loss, and other metrics of interest, and include a link to a website/info about inspecting for over/underfitting
- Plotting metrics – e.g. precision recall curve, ROC
- What to do next? how to really understand if you have trained a useful model, how to interpret results etc. how to use the model.
- get a feel for what you actually do when training an ML model – e.g. the typical size of your training set, the processes you did to assemble it (e.g. over and under sampling), typical number of epochs train for,

**Comment by sammlapp** [comment]
https://github.com/kitzeslab/annotation-notebooks start using this

**Comment by sammlapp** [comment]
I'm going to close this issue since we may develop extra documentation and tutorials, but they would not be part of opensoundscape
